ethereum/rpc/backend/backend.go

```diff
@@ -145,51 +145,49 @@ func (e *EVMBackend) EthBlockFromTendermint(
 	ethRPCTxs := []interface{}{}
 
 	for i, txBz := range block.Txs {
-		tx, gas := types.DecodeTx(e.clientCtx, txBz)
-
-		gasUsed += gas
-
-		msg, isEthTx := tx.(*evmtypes.MsgEthereumTx)
-
-		if !isEthTx {
-			// TODO: eventually support Cosmos txs in the block
-			continue
+		tx := types.DecodeTx(e.clientCtx, txBz)
+
+		for _, msg := range tx.GetMsgs() {
+			ethMsg, ok := msg.(*evmtypes.MsgEthereumTx)
+
+			if !ok {
+				continue
+			}
+
+			// Todo: gasUsed does not consider the refund gas so it is incorrect, we need to extract it from the result
+			gasUsed += ethMsg.GetGas()
+			hash := ethMsg.AsTransaction().Hash()
+			if !fullTx {
+				ethRPCTxs = append(ethRPCTxs, hash)
+				continue
+			}
+
+			// get full transaction from message data
+			from, err := ethMsg.GetSender(e.chainID)
+			if err != nil {
+				from = common.HexToAddress(ethMsg.From)
+			}
```

Comment:
I don't understand the error handling here

Reply:
its taken from original code. Should I remove and throw error?

Reply:
ok maybe logging the error would be useful

Reply:
sure, log added